projects/packages/blaze/src/class-dashboard-rest-controller.php

```diff
@@ -0,0 +1,270 @@
+<?php
+/**
+ * The Blaze Rest Controller class.
+ * Registers the REST routes for Blaze Dashboard.
+ *
+ * @package automattic/jetpack-blaze
+ */
+
+namespace Automattic\Jetpack\Blaze;
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use Automattic\Jetpack\Status\Host;
+use WP_Error;
+use WP_REST_Server;
+
+/**
+ * Registers the REST routes for Blaze Dashboard.
+ * It bascially forwards the requests to the WordPress.com REST API.
+ */
+class Dashboard_REST_Controller {
+	/**
+	 * Namespace for the REST API.
+	 *
+	 * @var string
+	 */
+	public static $namespace = 'jetpack/v4/blaze-app';
+
+	/**
+	 * Registers the REST routes for Blaze Dashboard.
+	 *
+	 * Blaze Dashboard is built from `wp-calypso`, which leverages the `public-api.wordpress.com` API.
+	 * The current Site ID is added as part of the route, so that the front end doesn't have to handle the differences.
+	 *
+	 * @access public
+	 * @static
+	 */
+	public function register_rest_routes() {
+		$site_id = $this->get_site_id();
+		if ( is_wp_error( $site_id ) ) {
+			return;
+		}
+
+		// WPCOM API routes
+		register_rest_route(
+			static::$namespace,
+			sprintf( '/sites/%d/blaze/posts', $site_id ),
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_blaze_posts' ),
+				'permission_callback' => array( $this, 'can_user_view_blaze_posts_callback' ),
+			)
+		);
+
+		// WordAds DSP API routes
+		register_rest_route(
+			static::$namespace,
+			sprintf( '/sites/%d/wordads/dsp/api/(?P<sub_path>.+)', $site_id ),
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_wordads_dsp_generic' ),
+				'permission_callback' => array( $this, 'can_user_view_wordads_dsp_callback' ),
+			)
+		);
+
+		register_rest_route(
+			static::$namespace,
+			sprintf( '/sites/%d/wordads/dsp/api/(?P<sub_path>.+)', $site_id ),
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'edit_wordads_dsp_generic' ),
+				'permission_callback' => array( $this, 'can_user_view_wordads_dsp_callback' ),
+			)
```

Comment:
Maybe we should add some checks for the `sub_path`, so folks do not input malicious things or paths that will not lead to a proper response when making the WordPress.com request?

Reply:
I wanted to have this open because we are making some changes on the DSP API and I didn't want to have Jetpack too wired to the endpoints names. 

My line of thought was that the API is exposed to the public, so users can make the same request they do here from a browser. 

But, if you want, I can add a whitelist of paths that we are allowed to call (maybe high-order paths like /campaigns, /posts, etc.) and leave the sub-path free to use (like /campaings/stats, etc.).

What do you think? If not I can go for an approach similar to the stats-admin and have each endpoint described and not use wildcards here.

Reply:
Thanks for the extra details! Since you're still iterating I can see how limiting ourselves would be problematic. I like the idea of a general whitelist though, that seems like a good middle-ground. 👍